docs/zh-cn/custom_quantum_functions.md

```diff
@@ -0,0 +1,491 @@
+# ��ζ�������̵Ĺ���
+
+���ںܶ����˵���ƻ����̿ɲ�ֻ������ĵ��Է����㰴���Ǹ����ô�򵥡���϶���ʵ�ֱȼ򵥰���ͺ���ӵĹ��ܡ�QMK��������ע�����Ĺ���, ���ǹ���, ���⣬�������Զ�������ڲ�ͬ����µ���Ϊ��
+
+��ҳ���ٶ��κ������QMK֪ʶ�����Ķ�[���QMK](understanding_qmk.md)�����ڸ��Ĳ��������ⷢ����ʲô��
+
+## A Word on Core vs ���� vs ����
+
+���ǰ�qmk��֯��һ���νṹ��
+
+* Core (`_quantum`)
+  * Keyboard/Revision (`_kb`)
+    * Keymap (`_user`)
+
+���������ÿһ��������ڶ����ϼ�һ��`_kb()`�� `_user()` ��׺�� �����ڼ���/�޶���ʹ��`_kb()`��׺���ڲ��ֲ�ʹ��`_user()`��׺��
+
+�ڼ���/�޶��㶨�庯��ʱ��`_kb()`��ִ���κδ���ǰ�ȵ���`_user()`�Ǳ�Ҫ�ģ���Ȼ���ֲ㺯��Ͳ�Ҫ�����á�
+<!-- �������⣺�����Ǿ䷭��Ĳ�̫��-->
+# �Զ������
+
+��ĿǰΪֹ�����������Ǹ�����м������Ϊ�򴴽��µļ��롣�Ӵ���Ƕ������Щ��������ơ�
+
+## ����һ���¼���
+
+���������һ������ö�ٳ���ȫ����Ҳ���Ǹ����������ֲ�����Ψһ��ֵ��QMKû��ֱ������������ֵ��С�������ṩ��һ��`SAFE_RANGE`�ꡣ�������ö��ʱ��`SAFE_RANGE`���֤��ȡ����Ψһ�ļ���ֵ��
+
+
+����ö������������ӡ������ӵ�`keymap.c`�Ļ�����ڲ���������`FOO`��`BAR`�ˡ�
+
+```c
+enum my_keycodes {
+  FOO = SAFE_RANGE,
+  BAR
+};
+```
+
+## Ϊ�������Ϊ���
+
+���㸲��һ���Ѵ��ڰ������Ϊʱ���������Ϊ�����¼�ʱ����Ҫ��`process_record_kb()`��`process_record_user()`������������ڼ�������ʵ���¼�������ǰ��QMK���á������������`true`��QMK��������ķ�ʽ������롣������Ժܷ�����չ����Ĺ��ܶ����滻��������`false` QMK����������Ȼ���ͼ���̧���ǰ����¼���������ˡ�
+
+��ĳ����»��ͷ�ʱ�������ᱻ���á�
+
+### process_record_user()`����ʾ��ʵ��
+
+���������������¡��Զ�����һ�����`FOO`�ļ������Ϊ�����������ڰ��»س�ʱ�������
+
+```c
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {s
```

Comment:
I don't think that S is supposed to be there.
 ```suggestion
  switch (keycode) {
```

Reply:
Ah, seems I must do my job quickly, I really need a new keyboard.
Thanks!